### PR TITLE
Tw updates: Round 1 spec to compliant - schema updates - orphans gone - non-compliant examples gone

### PR DESCRIPTION
**Pull request by m4sterbunny**
Tried various paths to get my diff. Is this the one?

### CODE REVIEW

**Review by m4sterbunny — COMMENTED** [review]
Highlighted some wip items in text (but not all)

**Review by jehrlich — APPROVED** [review]

File: src/openapi.yaml [hunk 1]
```diff
@@ -6,19 +6,18 @@ info:
...
     url: 'https://stage.VCPR.ORG'
-  - description: 'test server, uses regenerated test database'
+  - description: 'Test server, uses regenerated test database'
     url: 'https://test.VCPR.ORG'
-  - description: 'local docker container, uses regenerated test database'
+  - description: 'Local docker container, uses regenerated test database'
     url: 'http://localhost'
-
 security:
   - AuthToken: []
```
Comment (m4sterbunny) [inline src/openapi.yaml]: I think this can go, yes?

Comment (m4sterbunny) [reply src/openapi.yaml]: looks like a wee orphan

File: src/openapi.yaml [hunk 2]
```diff
@@ -473,20 +467,19 @@ paths:
...
           name: status
-          description: filter protocols to include only protocols with listed statuses. If this parameter
-            is omitted, the query defaults to all status values **except** DEPRECATED
+          description: 'filter protocols to include only protocols with listed statuses. If this parameter is omitted, the query defaults to all status values **except** DEPRECATED'
           schema:
             type: array
             items:
               $ref: '#/components/schemas/ProtocolStatus'
-          example: DEPRECATED|RECALLED
+          deprecated: true
```
[the remark this answers is not in the document]
Comment (jehrlich) [reply src/openapi.yaml:463]: DEPRECATED and RECALLED are valid status values
```
  object Status extends Enumeration {
    type Status = Value
    val PRESCRIBED,
    VALIDATED,
    SAVED, //initial state when created
    UNKNOWN,
    DEPRECATED,
    RECALLED
    = Value
```

File: src/openapi.yaml [hunk 3]
```diff
@@ -516,14 +498,17 @@ paths:
...
       operationId: updateProtocol
       tags:
         - editor
       summary: update protocol
-      description: update protocol
+      description: This endpoint updates a Protocol.
+      requestBody:
+        content:
+          application/json:
+            schema:
```
[the remark this answers is not in the document]
Comment (jehrlich) [reply src/openapi.yaml:508]: Yes, a Protocol is required in the POST body.

File: src/openapi.yaml [hunk 4]
```diff
@@ -986,48 +963,55 @@ components:
...
           type: string
+          example: testVOR
         userType:
           $ref: '#/components/schemas/UserType'
         email:
           type: string
           format: email
-          default: ''
+          example: testVOR@dairysight.com
         firstname:
```
[the remark this answers is not in the document]
Comment (jehrlich) [reply src/openapi.yaml:982]: yes

File: src/openapi.yaml [hunk 5]
```diff
@@ -986,48 +963,55 @@ components:
...
           type: string
+          example: Test
         lastname:
           type: string
+          example: Vet
         suffix:
           type: string
-      example: {"user":{"id":1,"uname":"testVOR","userType":"VOR","email":"testVOR@dairysight.com","firstname":"Test","lastname":"Vet","suffix":"DVM"}}
-
+          description: wip -- firstname and lastname can escape having a description, but what is this suffix?
```
[the remark this answers is not in the document]
Comment (jehrlich) [reply src/openapi.yaml]: Commonly "DVM"or  "VMD" here or "MRCVS" or "III". "Jr." i.e a degree or qualifier following the name

File: src/openapi.yaml [hunk 6]
```diff
@@ -1105,113 +1095,144 @@ components:
...
+        - ownerId
+        - name
+        - created
+        - edited
+        - status
+        - shareAll
       properties:
         id:
           type: integer
+          description: wip
```
[the remark this answers is not in the document]
Comment (jehrlich) [reply src/openapi.yaml:1157]: A common db idiom for an autoincremented key. Why troublesome?

File: src/openapi.yaml [hunk 7]
```diff
@@ -1105,113 +1095,144 @@ components:
...
+          description: User-friendly identifier for the protocol
+          example: example protocol
         shortname:
           type: string
+          description: Short name for the protocol
+          example: EXamp
         created:
           type: string
           format: date-time
+          description: wip is this ISO-*** Timestamp for protocol creation
```
[the remark this answers is not in the document]
Comment (jehrlich) [reply src/openapi.yaml]: yes ISO-8601

File: src/openapi.yaml [hunk 8]
```diff
@@ -1105,113 +1095,144 @@ components:
...
+          example: '2020-01-10T22:20:39Z'
         edited:
           type: string
           format: date-time
+          description: wip is this ISO-*** Timestamp for protocol update
```
[the remark this answers is not in the document]
Comment (jehrlich) [reply src/openapi.yaml]: yes

File: src/openapi.yaml [hunk 9]
```diff
@@ -1105,113 +1095,144 @@ components:
...
+          example: '2020-02-10T22:20:39Z'
         status:
           $ref: '#/components/schemas/ProtocolStatus'
         vetNote:
           type: string
+          description: wip
```
[the remark this answers is not in the document]
Comment (jehrlich) [reply src/openapi.yaml:1185]: I guess there should be a schema enumeration with values 

PRESCRIBED,
    VALIDATED,
    SAVED, 
    UNKNOWN,
    DEPRECATED,
    RECALLED

File: src/openapi.yaml [hunk 10]
```diff
@@ -1105,113 +1095,144 @@ components:
...
+          example: wip
         wtMeatHrs:
           type: integer
+          description: wip
```
[the remark this answers is not in the document]
Comment (jehrlich) [reply src/openapi.yaml:1189]: number of hours that meat may not be sold for food after last treatment

File: src/openapi.yaml [hunk 11]
```diff
@@ -1105,113 +1095,144 @@ components:
...
-
+          default: false
```
[the remark this answers is not in the document]
Comment (jehrlich) [reply src/openapi.yaml:1236]: correct

File: src/openapi.yaml [hunk 12]
```diff
@@ -2051,49 +2248,152 @@ components:
...
...shareAll":false,"shareClonable":false,"caseDefinition":{"pxId":0,"requireID":true,"requireMOR":true,"medical":"really sick","economic":"lotsa cash"},"treatments":[{"id":0,"ownerId":0,"description":"sample treatment","note":"this is an off-label Treatment","eligibilities":[{"id":1,"name":"Beef Cattle","fda":"BeefCattle"}],"tags":[{"id":1,"code":"EXAMPLE"}],"product":"MiracleMed","_dose":"3.0 ml per kg repeat in 6 hours and again in 24 hours","route":"intr-corporeal","sequenceFunction":{"intervals":["PT6H","PT24H"],"_type":"models.api.v1.SeqSeries"},"sizeFunction":{"dose":{"quantity":{"dimension":"Volume","value":3,"symbol":"ml"}},"perUnit":{"dimension":"Mass","value":1,"symbol":"kg"},"_type":"models.api.v1.DoseByWeightFunction"},"doseNote":"some note on dose","wtMeatHrs":22,"wtMilkHrs":5,"wtNote":"example"}],"tags":[{"id":1,"code":"EXAMPLE"}],"diagnoses":[{"id":8,"code":"PAIN","name":"Pain/inflammation (musculoskeletal), treatment"}],"eligibilities":[{"id":1,"name":"Beef Cattle","fda":"BeefCattle"}],"promises":[{"id":1,"name":"test user","code":"EXAMPLE"}]},"vorId":1,"morId":0,"refills":3,"status":"DEPRECATED","metadata":[],"rxId":"gG3pBCR1GCMKm89c","created":"2020-01-10T10:30:37Z"}
```
[the remark this answers is not in the document]
Comment (jehrlich) [reply src/openapi.yaml:2065]: Yes, optional. In effect, it defaults to 0, no refills.
```
case class Rx(
  protocol: Protocol,
  vorId: Long,
  morId: Long,
  refills: Option[Int] = None,
  expiration: Option[Instant],
  status: Rx.Status.Value,
  metadata: Seq[Map[String, String]] = Seq(),
  rxId: String = Random.alphanumeric.take(16).mkString,
  created: Instant = Instant.now().truncatedTo(ChronoUnit.SECONDS)
)
```

File: src/openapi.yaml (hunk 1 above)
Comment (m4sterbunny) [reply src/openapi.yaml]: @jehrlich I am referring to the security reference on line 23. IMO is serving no function -- yes?

File: src/openapi.yaml [hunk 13]
```diff
@@ -30,9 +29,7 @@ paths:
         - idempotent
       security: []
```
[the remark this answers is not in the document]
Comment (m4sterbunny) [reply src/openapi.yaml:28]: @jehrlich Just need your confirmation: Is this how the user gets their AuthToken and so security is genuinely null here?

File: src/openapi.yaml [hunk 14]
```diff
@@ -341,28 +335,33 @@ paths:
                         clonable:
```
[the remark this answers is not in the document]
Comment (m4sterbunny) [reply src/openapi.yaml:314]: closing in favour of this: https://github.com/VCPR-ORG/publicAPI/issues/17

File: src/openapi.yaml [hunk 15]
```diff
@@ -341,28 +335,33 @@ paths:
...
           schema:
             type: boolean
           required: false
+          explode: true
       responses:
         '204':
           description: No Content
-
     delete:
       operationId: unShareToAll
```
[the remark this answers is not in the document]
Comment (m4sterbunny) [reply src/openapi.yaml:346]: @jehrlich can you confirm please

File: src/openapi.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (m4sterbunny) [reply src/openapi.yaml:463]: @jehrlich can you confirm pls

File: src/openapi.yaml (hunk 3 above)
[the remark this answers is not in the document]
Comment (m4sterbunny) [reply src/openapi.yaml:508]: @jehrlich please confirm

File: src/openapi.yaml (hunk 13 above)
[the remark this answers is not in the document]
Comment (jehrlich) [reply src/openapi.yaml:28]: I am rewriting the security scheme spec and will submit a PR when that is ready for your review. Meanwhile you can ignore it.

File: src/openapi.yaml (hunk 15 above)
[the remark this answers is not in the document]
Comment (jehrlich) [reply src/openapi.yaml:346]: Let's delete the duplicate for now. Later we will review the server code and find undocumented endpoints.

**Review by jehrlich — APPROVED** [review]

File: src/openapi.yaml (hunk 1 above)
Comment (m4sterbunny) [reply src/openapi.yaml]: self-approving and deleting

**Review by m4sterbunny — COMMENTED** [review]
items resolved or issues raised to allow merge

**Pull request merged**